### PR TITLE
SP-4013: Job failed playbook

### PR DESCRIPTION
**Pull request by vic3lord**

### CODE REVIEW

File: kubernetes/job-failed/playbook.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            POD_NAME: '{{ .value.name }}'
+            SP_FORMATTER: "raw"
+            ALL_CONTAINERS: "true"
+
+        - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+          envFrom:
+            integrationRef: "{{ $.params.slack_integration }}"
+          env:
+            CONTENT: "{{ .output }}"
+            RECIPIENTS: "{{ .params.recipient }}"
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:48]:
```suggestion
            RECIPIENTS: "{{ $.params.recipient }}"
```

File: kubernetes/job-failed/playbook.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            NAMESPACE: '{{ $.params.namespace }}'
+            POD_NAME: '{{ .value.name }}'
+            SP_FORMATTER: "raw"
+            ALL_CONTAINERS: "true"
+
+        - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+          envFrom:
+            integrationRef: "{{ $.params.slack_integration }}"
+          env:
+            CONTENT: "{{ .output }}"
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:47]:
```suggestion
            CONTENT: "{{ $.output }}"
```

File: kubernetes/job-failed/playbook.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      SP_REDIRECT_URL: https://app.stackpulse.io/execution/{{ .execution.id }}
+      HEADER_MESSAGE: Hello
+      RECIPIENTS: "{{ $.params.recipient }}"
+      QUESTION: "Delete job {{ $.params.job_name }}?"
+      RESPONSES_JSON: '{"No": "no", "Yes": "yes"}'
+      MESSAGE_TYPE: "BUTTONS_SELECT"
+      TIMEOUT: "30m"
+      DEFAULT_RESPONSE: "Yes"
+
+  - when: '{{ eq .slack_response "yes" }}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:65]:
```suggestion
  - when: '{{ eq $.slack_response "yes" }}'
```

File: kubernetes/job-failed/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Kubernetes job failed
+
+<img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png" width="100">
+
+![env](https://img.shields.io/static/v1?label=env&message=Kubernetes&style=flat&logo=Kubernetes&color=326CE5)
+![uses-slack](https://img.shields.io/static/v1?label=uses&message=Slack&style=flat&logo=slack&color=4A154B)
+
+[![Import to StackPulse](../../images/open_in_stackpulse.svg)](https://app.stackpulse.io/playbook/create#https://github.com/stackpulse/playbooks/blob/master/kubernetes/job-failed/playbook.yaml)
+
+This playbook will help you extract the logs from a failed job and help you optionally delete it once you understand the root cause.
```
Comment (alongat) [inline kubernetes/job-failed/README.md:10]: Looks like a long sentence, please send the texts through product

Comment (vic3lord) [reply kubernetes/job-failed/README.md:10]: Sure thing

**Review by alongat — APPROVED** [review]
Please ask @rzsp to go over the text

File: kubernetes/job-failed/playbook.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+        - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+          envFrom:
+            integrationRef: "{{ $.params.slack_integration }}"
+          env:
+            CONTENT: "{{ $.output }}"
+            RECIPIENTS: "{{ $.params.recipient }}"
+            SP_REDIRECT_URL: https://app.stackpulse.io/execution/{{ .execution.id }}
+
+  - name: gcr.io/stackpulse/steps/slack/send_interactive_message
```
Comment (eldadru) [inline kubernetes/job-failed/playbook.yaml:51]: This is the old step, please upgrade to use the new one (gcr.io aren't supported anymore)

**Review by rzsp — CHANGES_REQUESTED** [review]

File: kubernetes/job-failed/playbook.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    id: Ask permission to delete
+    envFrom:
+      integrationRef: "{{ $.params.slack_integration }}"
+    env:
+      SP_REDIRECT_URL: https://app.stackpulse.io/execution/{{ .execution.id }}
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:56]:
```suggestion
      SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
```

File: kubernetes/job-failed/playbook.yaml (hunk 5 above)
Comment (vic3lord) [inline kubernetes/job-failed/playbook.yaml:51]:
```suggestion
  - name: us-docker.pkg.dev/stackpulse/public/slack/message/interactive
```

File: kubernetes/job-failed/playbook.yaml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            SP_FORMATTER: "raw"
+            ALL_CONTAINERS: "true"
+
+        - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+          envFrom:
+            integrationRef: "{{ $.params.slack_integration }}"
+          env:
+            CONTENT: "{{ $.output }}"
+            RECIPIENTS: "{{ $.params.recipient }}"
+            SP_REDIRECT_URL: https://app.stackpulse.io/execution/{{ .execution.id }}
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:49]:
```suggestion
           SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
```

File: kubernetes/job-failed/playbook.yaml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  - name: us-docker.pkg.dev/stackpulse/public/slack/message/interactive
+    id: Ask permission to delete
+    envFrom:
+      integrationRef: "{{ $.params.slack_integration }}"
+    env:
+      SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
+      HEADER_MESSAGE: Hello
+      RECIPIENTS: "{{ $.params.recipient }}"
+      QUESTION: "Delete job {{ $.params.job_name }}?"
+      RESPONSES_JSON: '{"No": "no", "Yes": "yes"}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:60]: I think this changed to No, Yes

File: kubernetes/job-failed/playbook.yaml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            SP_FORMATTER: "raw"
+            ALL_CONTAINERS: "true"
+
+        - name: us-docker.pkg.dev/stackpulse/public/slack/message/snippet
+          envFrom:
+            integrationRef: "{{ $.params.slack_integration }}"
+          env:
+            CONTENT: "{{ $.output }}"
+            RECIPIENTS: "{{ $.params.recipient }}"
+           SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:49]:
```suggestion
            SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
```

File: kubernetes/job-failed/playbook.yaml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+  - foreach:
+      in: filtered
+      key: key
+      value: value
+      steps:
+        - name: us-docker.pkg.dev/stackpulse/public/kubectl/logs
+          runner: "{{ $.params.runner }}"
+          env:
+            NAMESPACE: '{{ $.params.namespace }}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:38]:
```suggestion
            NAMESPACE: "{{ $.params.namespace }}"
```

File: kubernetes/job-failed/playbook.yaml [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+            POD_NAME: '{{ .value.name }}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:39]:
```suggestion
            POD_NAME: "{{ .value.name }}"
```

File: kubernetes/job-failed/playbook.yaml [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    type: integration
+    description: Slack integration
+    options:
+      integration_type: slack
+steps:
+  - name: us-docker.pkg.dev/stackpulse/public/kubectl/get-pods
+    id: get-pods
+    runner: "{{ $.params.runner }}"
+    env:
+      NAMESPACE: '{{ $.params.namespace }}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:26]:
```suggestion
      NAMESPACE: "{{ $.params.namespace }}"
```

File: kubernetes/job-failed/playbook.yaml [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+      NAME_CONTAINS: '{{ $.params.job_name }}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:27]:
```suggestion
      NAME_CONTAINS: "{{ $.params.job_name }}"
```

**Review by alongat — CHANGES_REQUESTED** [review]
See small ones

File: kubernetes/job-failed/playbook.yaml [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+  - name: us-docker.pkg.dev/stackpulse/public/slack/message/interactive
+    id: Ask permission to delete
+    envFrom:
+      integrationRef: "{{ $.params.slack_integration }}"
+    env:
+      SP_REDIRECT_URL: {{ $.metadata.executionUrl }}
+      HEADER_MESSAGE: Hello
+      RECIPIENTS: "{{ $.params.recipient }}"
+      QUESTION: "Delete job {{ $.params.job_name }}?"
+      RESPONSES_JSON: '{"No": "no", "Yes": "yes"}'
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:60]:
```suggestion
      RESPONSES: "No, Yes"
```

File: kubernetes/job-failed/playbook.yaml [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+      MESSAGE_TYPE: "BUTTONS_SELECT"
```
Comment (alongat) [inline kubernetes/job-failed/playbook.yaml:61]:
```suggestion
      SELECTOR_TYPE: "BUTTONS_SELECT"
```

**Review by alongat — APPROVED** [review]

**Pull request merged**